src/error.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ *  Contact: Lukasz Pawelczyk <l.pawelczyk@samsung.com>
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License
+ */
+
+use std::error;
+use std::fmt;
+
+
+/// Enumeration of YACA error values returned from the C library
+///
+/// - All of the API functions return [`Result`] with an `Error`
+///   embedded.
+/// - If `Result::Err` returned it will contain one of those values.
+/// - They are passed directly from the C `YACA` implementation.
```

Comment:
How the hell should I know why? Cause I felt like? Cause my fingers led me to this solution? Cause I didn't know any better at that time? Cause I had a plan to conquer the world and this was just a beginning but then I failed? Cause I had a hangover? Cause my other self was in control at that point? Cause I wanted to see if you notice? Because we'll all die in the end anyway?

Reply:
The second to last possibility I find.. unsettling.

Reply:
Don't worry, it was only one test of several, you failed anyway. At least that's what the other self tells me. I still don't know if I can trust him though.